### DISCUSSION TITLE
Discussion: Collection handling in vector database

### DISCUSSION THREAD

**Discussion by jonathan-rohde** [open-webui/open-webui#3427] [General]
I want to raise the following discussion.

Status quo:
- One document results in one collection in the vector database
- The hash of the file content is the collection name
- The meta data contains the filename
- Deletion of documents keep data still in vector database

Questions:
- What do you think of storing all documents in a single collection? 
  - My opinion: A vector database is more efficient finding the k nearest vectors from one single collection than a python application fetching k nearest vectors from x collections, sorting them and taking first k vectors from result.
  Filtering based on meta data can easily be implemented to limit a search based on the file hash.
  Please correct me, if my assumption is incorrect.

- Shouldn't the vectors get deleted in the vector store, once the document is deleted?
  - I can see the benefit of having it still in the store as if the file is scanned/uploaded again, it an just be reused. 
  But I see a problem, when for example we want to add or update meta data, like the file name.

**Comment by GridexX** [comment]
Hello @jonathan-rohde and everyone else !

I completely align with your thoughts, Jonathan. In my current role, I am working on a research project where scientists need to efficiently search through a collection of documents. Given this use case, I believe that the points you raised are highly relevant and important to consider.

Vector databases are specifically designed to find the nearest documents, so it seems natural to leverage this capability for our requirements. In my opinion, utilizing the query finder for Chroma would be a more effective approach to handle this task. I am eager to hear more perspectives on this matter from the rest of the community.

To further explore this idea, I plan to open an issue dedicated to this topic. This will allow us to gather additional insights and potentially refine our approach based on the collective expertise of the community.

Looking forward to continuing this discussion and working together to find the best solution.
